include/micm/solver/lu_decomposition.inl

```diff
@@ -316,15 +312,25 @@ namespace micm
             if (U_vector[uii_deref + i_cell] == 0.0)
             {
               is_singular = true;
-              return;
             }
             L_vector[lki_nkj_first + i_cell] /= U_vector[uii_deref + i_cell];
           }
           ++lki_nkj;
           ++uii;
         }
       }
+      std::size_t uii_deref = *uii;
+      if (n_cells != A_GroupVectorSize) {
+        // check the bottom right corner of the matrix
+        for (std::size_t i_cell = 0; i_cell < n_cells; ++i_cell)
+        {
+          if (U_vector[uii_deref + i_cell] == 0.0)
+          {
+            is_singular = true;
+          }
```

Comment:
Does it make sense to break the loop after setting the singularity flag?

```c++
        for (std::size_t i_cell = 0; i_cell < n_cells; ++i_cell)
        {
          if (U_vector[uii_deref + i_cell] == 0.0)
          {
            is_singular = true;
            break;
          }
        }
```

Reply:
The `break` is removed here to allow a user bypass the singularity check even if the matrix is singular, just like what the current MICM is doing.

Reply:
hmm I don't get it. The loop just sets `is_singular` true or false. If it sets true at an early iteration, it doesn't matter whether continuing checking and setting it. Can you explain how it is related to allowing users to bypass the check?

Reply:
If you `break` here, the LU decomposition will end up with incompletion. If a user wants to bypass singularity check by setting `parameters_.check_singularity_ = false` 
https://github.com/NCAR/micm/blob/fc66c175a5a7857a7e2c76cf1726d6e55855b8c3/include/micm/solver/rosenbrock.inl#L255
the solver will proceed with wrong LU results and probably blow up later.

Reply:
hmm, is there any chance you might be looking at the different line for singularity checks? It should be true for [the above check](https://github.com/NCAR/micm/blob/572-check-for-singularity-when-the-solver-parameters-flag-is-turned-on/include/micm/solver/lu_decomposition.inl#L309-L316) but doesn't seem like it for this check

Reply:
Oh my bad. I am looking at the wrong section of the code. I think you are right here and sorry for the confusion.

Reply:
This particular one is fine to break at since it will only stop the singularity check for one sub block of the matrix. The whole LU decomposition will continue on though